composables/store/project.ts

```diff
@@ -115,28 +119,37 @@ export const useProjectStore = defineStore('project', () => {
           selectedN = addOrRemove(objectId, false);
         });
       }
+    }
+
+    // For each object selected, loop through the selected objects
+    selectedN = R.pickBy((_, objectId): boolean => {
+      const object = getObject.value(objectId);
+
+      // If addToAllowChoice is true, add the number of additional choices to allowedChoices
+      if (object.addToAllowChoice && object.idOfAllowChoice === rowId) {
+        allowedChoices = allowedChoices + object.numbAddToAllowChoice;
+      }
 
-      // Remove any objects that are incompatible with the selected object
-      selectedN = R.pickBy((_, objectId): boolean => {
-        const object = getObject.value(objectId);
+      if (isSelected) {
```

Comment:
This was moved to check inside the `pickBy` as we only want to loop over the same data once, but want to check if the allowed choices were modified more than once.